### PR TITLE
fix: missing items that match after update/patch

### PR DESCRIPTION
**Pull request by mariusheine**

### CODE REVIEW

File: src/useFind.ts [hunk 1]
```diff
@@ -39,13 +39,19 @@ function loadServiceEventHandlers<
...
+        existing = true;
         return changedItem;
       }
 
       return item;
     });
+
+    if (!existing) {
+      data.value = [...data.value, changedItem];
+    }
```
Comment (lukashass) [inline src/useFind.ts]: @mariusheine @anbraten What do you think of this instead?

```suggestion
    const itemIndex = data.value.findIndex((item) => getId(item) === getId(changedItem));
    if (itemIndex === -1) {
      data.value = [...data.value, changedItem];
    } else {
      data.value = [...data.value.slice(0, itemIndex), changedItem, ...data.value.slice(itemIndex + 1)];
    }
```

Comment (mariusheine) [reply src/useFind.ts]: @lukashass I really like it!

@anbraten ?

Comment (anbraten) [reply src/useFind.ts]: LGTM. It keeps the order in case the item already exists.

Comment (mariusheine) [reply src/useFind.ts]: @anbraten @lukashass I applied the suggestion. Can one of you please approve then?

**Review by anbraten — CHANGES_REQUESTED** [review]
Could you add a check to some existing test maybe behind line 514 to check that the changed item is still at the same position and not only contained in the array. That is quite important to prevent list from changing their orders after receiving an update.

**Comment by mariusheine** [comment]
> Could you add a check to some existing test maybe behind line 514 to check that the changed item is still at the same position and not only contained in the array. That is quite important to prevent list from changing their orders after receiving an update.

@anbraten I added a test. Please have a look

**Review by anbraten — APPROVED** [review]

**Pull request merged**